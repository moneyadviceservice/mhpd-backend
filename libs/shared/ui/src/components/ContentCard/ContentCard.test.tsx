import React from 'react';
import { render, screen } from '@testing-library/react';
import { ContentCard } from '.';

describe('ContentCard component', () => {
  it('renders correctly', () => {
    render(
      <ContentCard
        title={'test title'}
        image={{
          src: '/images/next-steps-control.png',
          alt: 'some alt text',
        }}
      >
        Test content
      </ContentCard>,
    );
    const contentCard = screen.getByTestId('content-card');
    expect(contentCard).toMatchSnapshot();
  });

  it('renders correctly with no image source', () => {
    render(<ContentCard title={'test title'}>Test content</ContentCard>);
    const contentCard = screen.getByTestId('content-card');
    expect(contentCard).toMatchSnapshot();
  });
});
