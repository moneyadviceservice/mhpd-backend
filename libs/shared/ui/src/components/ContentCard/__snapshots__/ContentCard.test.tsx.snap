// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCard component renders correctly 1`] = `
<div
  class="md:min-h-[440px] border-1 border-slate-400 rounded-bl-[36px] w-[408px] shadow-bottom-gray print:border-none print:shadow-none print:w-full"
  data-testid="content-card"
>
  <img
    alt="some alt text"
    class="print:hidden"
    data-nimg="1"
    decoding="async"
    height="204"
    loading="lazy"
    src="/_next/image?url=%2Fimages%2Fnext-steps-control.png&w=828&q=75"
    srcset="/_next/image?url=%2Fimages%2Fnext-steps-control.png&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fnext-steps-control.png&w=828&q=75 2x"
    style="color: transparent;"
    width="408"
  />
  <h3
    class="text-lg md:text-xl text-gray-800 font-bold m-4"
  >
    test title
  </h3>
  <div
    class="mx-4 pb-4 text-lg text-gray-800"
  >
    Test content
  </div>
</div>
`;

exports[`ContentCard component renders correctly with no image source 1`] = `
<div
  class="md:min-h-[440px] border-1 border-slate-400 rounded-bl-[36px] w-[408px] shadow-bottom-gray print:border-none print:shadow-none print:w-full"
  data-testid="content-card"
>
  <h3
    class="text-lg md:text-xl text-gray-800 font-bold m-4"
  >
    test title
  </h3>
  <div
    class="mx-4 pb-4 text-lg text-gray-800"
  >
    Test content
  </div>
</div>
`;
